content/js/record.md

```diff
@@ -1,307 +1,331 @@
+skygear.publicDB.delete({
+  id: 'note/f705fd10-bfd8-47dd-aec9-3e3de32a8f5a'
+}).then((deletedRecord) => {
+  console.log(deletedRecord);
+}).catch((error) => {
+  console.error(error);
 });
-const author = new skygear.Reference(rick);
-note.author = author;
-skygear.publicDB.save(note);
+
 ```
 
-You can build up reference between records.
+Deleting multiple records:
+
 
 ``` javascript
-const note1 = new Note({
-  heading: 'Specification',
-  content: 'This is first section',
-});
-const note2 = new Note({
-  heading: 'Specification page 2',
-  content: 'This is second section',
-});
-note1.nextPage = new skygear.Reference(note2);
-skygear.publicDB.save([note2, note1]);
-```
+const blogQuery = new skygear.Query(Blog)
+  .lessThan('rating', 3);
+
+skygear.publicDB.query(blogQuery)
+  .then((blogs) => {
+    console.log(blogs);
+    // [blog1, blog2, blog3]
+    return skygear.publicDB.delete(notes);
+  })
+  .then((errors) => {
+    console.log(errors);
+    // [undefined, undefined, error3]
+  })
+  .catch((error) => {
+    // request error
+  });
 
-- Ordering of objects in batch save array matters if there are reference
-relationship between records. In the above case, if you perform batch save on
-`[note1, note2]`, there will be an error saving `note1`, but `note2` will
-still be saved!
-- If you wish to retrieve `note1` and `note2` at the same time in one query,
-you might use `transientInclude`. Read the [Queries](/js/guide/query#relational-queries)
-section to learn more about eager loading.
+```
 
-### Deleting Referenced Record
+If some delete operation fails (e.g. due to insufficient permission) the error
+message(s) will be reflected in the errors array illustrated above.
 
-Yet to be implemented. For now, you have to delete the referencing record
-first and then the referenced record.
 
 <a name="data-type"></a>
-## Data Type
+## Data Types
+
+Skygear supports most of the built-in JavaScript types:
 
-Skygear supports almost all of the builtin JavaScript types, such as:
 - String
 - Number
 - Boolean
 - Array
 - Object
 - Date
 
-There are also four other types provided by Skygear JS SDK:
-- Reference (described above)
-- Sequence (described below)
-- Geolocation (described in [Geo-location](/js/guide/geolocation) section)
-- Asset (described in [Assets (File Upload)](/js/guide/asset) section)
+In addition, Skygear JS SDK 4 more types:
+
+- [Reference](#reference)
+- [Sequence](#sequence)
+- [Geo Location](/js/guide/geolocation)
+- [Assets](/js/guide/asset) (File Uploads)
+
+Refer to the [server](/server/guide/data-type) documentation for
+more detail on supported data types.
+
+
+<a name="reference"></a>
+## Records Relations
+
+Skygear supports relation between records via _references_.
+`skygear.Reference` is a class which will translate to foreign keys in
+the skygear database for efficient queries.
+
+``` javascript
+const Message = skygear.Record.extend('message');
+const Person = skygear.Record.extend('person');
```

Comment:
you can also mention that "Imagine we have two entities: Message and Person, where a message record has a sender `from` and a receiver `to`, both being a Person."